src/MoBi.Engine/Sbml/UnitDefinitionImporter.cs

```diff
@@ -48,37 +53,39 @@ public class UnitDefinitionImporter : SBMLImporter, IStartable, IUnitDefinitionI
       private readonly IMoBiDimensionFactory _moBiDimensionFactory;
       private readonly IDictionary<int, Unit> _baseUnitsDictionary;
-      private readonly IDictionary<string, UnitConvertionInfo> _unitConvertionDictionary;
-      private IDimensionFactory _dimensionFactory;
-      public IReadOnlyDictionary<string, IDimension> ConvertionDictionary { get => _unitConvertionDictionary.ToDictionary(kv => kv.Key, kv => kv.Value.Dimension); }
-      private IDictionary<string, string> _sbmlUnitsSynonyms = new Dictionary<string, string>()
+      private readonly IDictionary<string, UnitConversionInfo> _unitConversionDictionary;
+
+      public IReadOnlyDictionary<string, IDimension> ConversionDictionary => _unitConversionDictionary.ToDictionary(kv => kv.Key, kv => kv.Value.Dimension);
+
+      private readonly IDictionary<string, string> _sbmlUnitsSynonyms = new Dictionary<string, string>()
       {
-         { "litre", "l" }
+         {"litre", "l"}
       };
 
-      public UnitDefinitionImporter(IObjectPathFactory objectPathFactory, IObjectBaseFactory objectBaseFactory, IMoBiDimensionFactory mobiDimensionFactory, ASTHandler astHandler, IMoBiContext context, IDimensionFactory dimensionFactory) : base(objectPathFactory, objectBaseFactory, astHandler, context)
+      public UnitDefinitionImporter(IObjectPathFactory objectPathFactory, IObjectBaseFactory objectBaseFactory, IMoBiDimensionFactory mobiDimensionFactory, ASTHandler astHandler, IMoBiContext context) : 
+         base(objectPathFactory, objectBaseFactory, astHandler, context)
       {
          _moBiDimensionFactory = mobiDimensionFactory;
          _baseUnitsDictionary = new Dictionary<int, Unit>();
-         _unitConvertionDictionary = new Dictionary<string, UnitConvertionInfo>();
-         _dimensionFactory = dimensionFactory;
-         InitBaseUnitsDictionary();
+         _unitConversionDictionary = new Dictionary<string, UnitConversionInfo>();
+         initBaseUnitsDictionary();
       }
 
       public string TranslateUnit(string sbmlUnit)
       {
          if (_sbmlUnitsSynonyms.ContainsKey(sbmlUnit))
             return _sbmlUnitsSynonyms[sbmlUnit];
+
          return sbmlUnit;
       }
 
-      private void InitBaseUnitsDictionary()
+      private void initBaseUnitsDictionary()
       {
          _baseUnitsDictionary.Add(libsbml.UNIT_KIND_AMPERE, _moBiDimensionFactory.Dimension("Ampere").Unit("A"));
          _baseUnitsDictionary.Add(libsbml.UNIT_KIND_BECQUEREL, _moBiDimensionFactory.Dimension("Becquerel").Unit("Bq"));
          _baseUnitsDictionary.Add(libsbml.UNIT_KIND_CANDELA, _moBiDimensionFactory.Dimension("Candela").Unit("cd"));
          _baseUnitsDictionary.Add(libsbml.UNIT_KIND_COULOMB, _moBiDimensionFactory.Dimension("Coulomb").Unit("C"));
-         _baseUnitsDictionary.Add(libsbml.UNIT_KIND_DIMENSIONLESS, _moBiDimensionFactory.NoDimension.DefaultUnit);
+         _baseUnitsDictionary.Add(libsbml.UNIT_KIND_DIMENSIONLESS, _moBiDimensionFactory.NoDimension.BaseUnit);
```

Comment:
I am pretty sure you mean BaseUnit and not DefaultUnit. We always need to convert to BaseUnit.